src/helper/Backend.cpp

```diff
@@ -70,6 +75,26 @@ namespace SDDM {
                         .arg(mainConfig.X11.UserAuthFile.get());
                 env.insert(QStringLiteral("XAUTHORITY"), value);
             }
+#if defined(Q_OS_FREEBSD)
+        /* get additional environment variables via setclassenvironment();
+            this needs to be done here instead of in UserSession::setupChildProcess
+            as the environment for execve() is prepared here */
+        login_cap_t *lc;
+
+        if (lc = login_getpwclass(pw)) {
+            setclassenvironment(lc, pw, 1);     /* path variables */
+            setclassenvironment(lc, pw, 0);     /* non-path variables */
+            login_close(lc);
+            if (lc = login_getuserclass(pw)) {
+                setclassenvironment(lc, pw, 1);
+                setclassenvironment(lc, pw, 0);
+            }
+            if (lc)
+                login_close(lc);
+            /* copy all environment variables that are now set */
+            env.insert(QProcessEnvironment::systemEnvironment());
```

Comment:
Are you sure the environment is clean(-ish) before calling `setclassenvironment()` and then inserting the whole thing here? The implementation in `setclassenvironment()` just calls `setenv()` so you're potentially pulling in the whole environment of SDDM.

Reply:
I have pulled another revision of this patch from https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=244341